From fc27f8600dfd781691f50d62a46166d6af175d02 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Fri, 2 Jun 2017 11:03:21 -0700
Subject: [PATCH] MANTA-3288 cannot preview operator-guide in build/docs/public
 before publishing to gh-pages Reviewed by: David Pacheco <dap@joyent.com>

---
 .gitignore                                          | 4 ++--
 Makefile                                            | 8 +++++---
 docs/{operator-guide/index.md => operator-guide.md} | 0
 3 files changed, 7 insertions(+), 5 deletions(-)
 rename docs/{operator-guide/index.md => operator-guide.md} (100%)

diff --git a/.gitignore b/.gitignore
index 5a67096..6b5cff1 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,7 +1,7 @@
 /tmp
 /build
-/docs/operator-guide/*.html
-/docs/operator-guide/*.json
+/docs/operator-guide.html
+/docs/operator-guide.json
 /docs/user-guide/examples/*/job.id
 /docs/user-guide/examples/*/job.err
 /docs/user-guide/examples/*/job.out
diff --git a/Makefile b/Makefile
index c8c596a..eae89b9 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@
 #
 
 #
-# Copyright (c) 2016, Joyent, Inc.
+# Copyright (c) 2017, Joyent, Inc.
 #
 
 #
@@ -15,7 +15,7 @@
 # Only care to build the operator guide for publishing directly out of
 # this repo (to <http://joyent.github.io/manta/>). The other docs are
 # pulled into apidocs.joyent.com.git for publishing there. See RFD 23.
-DOC_FILES	 = operator-guide/index.md
+DOC_FILES	 = operator-guide.md
 
 include ./tools/mk/Makefile.defs
 
@@ -29,6 +29,7 @@ all: docs
 docs-regenerate-examples:
 	docs/user-guide/examples/regenerate-all-examples.sh
 
+CLEAN_FILES += docs/operator-guide.{html,json} build/docs
 
 # Update the operator guide at <http://joyent.github.io/manta/>.
 .PHONY: publish-operator-guide
@@ -40,7 +41,8 @@ publish-operator-guide: docs
 	mkdir -p tmp
 	[[ -d tmp/gh-pages ]] || git clone git@github.com:joyent/manta.git tmp/gh-pages
 	cd tmp/gh-pages && git checkout gh-pages && git pull --rebase origin gh-pages
-	cp build/docs/public/operator-guide/index.html tmp/gh-pages/index.html
+	rsync -av build/docs/public/media/ tmp/gh-pages/media/
+	cp build/docs/public/operator-guide.html tmp/gh-pages/index.html
 	(cd tmp/gh-pages \
 		&& git commit -a -m "$(MSG)" \
 		&& git push origin gh-pages || true)
diff --git a/docs/operator-guide/index.md b/docs/operator-guide.md
similarity index 100%
rename from docs/operator-guide/index.md
rename to docs/operator-guide.md
-- 
2.21.0

